exporter/datadogexporter/translate_traces.go

```diff
@@ -630,3 +634,27 @@ func eventsToString(evts pdata.SpanEventSlice) string {
 	eventArrayBytes, _ := json.Marshal(&eventArray)
 	return string(eventArrayBytes)
 }
+
+func generateSpanNameMap(cfg *config.Config) map[string]string {
+	// we can preset the size to be at most the number of remapping entries
+	spanNameMap := make(map[string]string, len(cfg.Traces.SpanNameRemappings))
+	for _, entry := range cfg.Traces.SpanNameRemappings {
+		spanNameRemapTuple := strings.Split(entry, " ")
+		spanNameMap[spanNameRemapTuple[0]] = spanNameRemapTuple[1]
+	}
+
+	return spanNameMap
+}
+
+// remapDatadogSpanName allows users to map their datadog span operation names to
+// another string as they see fit.  It's unclear if this should be a set of regexes or simple string
+// matching. It's also unclear if we should expose arbitrary remapping, ie, allowing users to map
+// to some other dyanmic attribute or value on the span. This would probably cause issues in metric
+// computation
```

Comment:
We should describe what the function does, I think the design comment while valid does not belong here
```suggestion
// remapDatadogSpanName allows users to map their datadog span operation names to
// another string as they see fit.
```

Does the current design allow for extending this for regexes or arbitrary remapping?

Reply:
we do regex for `ignoreResoources` so it wouldn't be too hard to apply that way, i am a bit concerned that, because span operation names have pretty strict requirements for needing to be a static/consistent value per `service/IL Library/span kind` a regex could end up introducing lots of unintended edge cases, ie, folks providing too broad a regex match and combining all span names into 1. Let me think though this a bit more though, I think regex has some benefits as well.